### ISSUE TITLE
VTO Widget provide

### ISSUE THREAD

**Issue by whilemouse** [jeeliz/jeelizFaceFilter#60]
Hi, @xavierjs 
I read [this](https://github.com/jeeliz/jeelizFaceFilter/issues/56#issuecomment-435498633).

>If you want to do glasses VTO, we can discuss about it. Although our current VTO solution is not very open, we can discuss that. I think we can provide a very good solution for a very honest price. Our VTO app also reconstructs the lighting (both ambient and directionnal) and we have an image based server side fallback for people who are not compatible with WebGL.

I want to buy VTO solution and customize.
1. How to buy VTO solution?
2. Can I use a model that was created only with the webgl-based rendering program? For example, can I use models created with vray, keyshot, marmoset instead of webgl?
3. If I can not do it 2, can I put the model I created in webgl? ex) .gltf(material was included) or .json(material made of three js)
4. What features do you offer if you buy a solution?

**Comment by xavierjs** [comment]
Hi @whilemouse 

The solution includes the modelisation.
Our architecture is the following:

## Production server:

* a server providing a database service called glassesDB.
  The client app (the widget or https://jeeliz.com/sunglasses) directly requests glassesDB in order to get the 3D models URL, the prices, the metadata of the glasses
glassesDB is the data reference. Each model of glassesDB is identified by a SKU ( https://github.com/jeeliz/jeelizGlassesVTOWidget )
* some crawler daemons are linked to glassesDB to enrich the model metadata. For example a crawler get the prices, the reviews and the images from amazon, another from aliexpress, ...


## Infography preprod server:

On our preprod server, the 3D infographist will:

* upload by FTP the 3D model in OBJ file format (the OBJ should respect some constraints, not all OBJ are accepted. However the rules are quite easy for a 3D infographist)
* Browse uploaded models, inspect and test this particular model using a web app called the adjuster. He will also change a bit the scale, the position, the rotation... In order to avoid to waste time by coming back to its 3D modeling software, re-upload it, re-test it, ...
* Then he will declare the materials using the material editor. It uses our own PBR 3D engine, not THREE.JS. This is very specific.
* Then put together materials with meshes using another app to create the glasses models. At this point glasses are grouped by stores.
* Then export the 3D models and generate the thumbnails on the fly with another app.

This workflow is pretty smooth internally, so we can currently deliver a service with a fixed integration cost, a price per model and a price per 1000 VTO sessions where only the VTO widget will run on the client server. You can ask for a more detailed pricing by contacting Chouki at contact__at__jeeliz.com (our prices are still cheaper than the main other VTO solutions).

But this solution is not packaged enough to be sold externally. It is still very doable to run a second instance of glassesDB on another server, but the glasses building workflow is not ready to be externalized.

But alternative solutions are coming for a standalone VTO solution:

* FaceFilter will be improved in the next few months (I know this is possible to get a better positionning)
* We are going to release FaceFilterLM, which is a landmark based version of facefilter.
The positionning will be more accurate since it uses precise facial landmark (around the ears if the user is rotating the head left<->right) and we hope to propose a very good VTO solution with an integration sample with THREE.JS. The library will not be free, but its cost will be lower than elsewhere and there will be FaceFilter bindings (so an app working with facefilter will be quickly adapted to FaceFilterLM).

**Comment by whilemouse** [comment]
Thanks. @xavierjs 
Is there a trial version? I want to experience this.

**Comment by xavierjs** [comment]
Hi,

Unfortunately there is no trial version. Since last January we have changed our strategy, we used to focus on the sunglasses VTO use case and we have not done many developments since. At this time we focused more on glasses distributors which are networks of shops, then on glasses makers who often have e-commerce websites. They are looking on integrated solutions where the glasses modelisation is provided (either they don't have the 3D models, or it is CAD models, too high poly for real time 3D rendering). That's why we did these choices.
